src/prog_models/prognostics_model.py

```diff
@@ -1272,13 +1278,19 @@ def optimization_fcn(params):
         
         params = np.array([self.parameters[key] for key in keys])
 
-        res = minimize(optimization_fcn, params, method=method, bounds = config['bounds'], options=config['options'])
+        res = minimize(optimization_fcn, params, method=method, bounds = config['bounds'], options=config['options'], tol=tol)
+
+        if not res.success:
+            warn(f"Parameter Estimation did not converge: {res.message}")
+
         for x, key in zip(res.x, keys):
             self.parameters[key] = x
-
+        
         # Reset noise
         self.parameters['measurement_noise'] = m_noise
-        self.parameters['process_noise'] = p_noise   
+        self.parameters['process_noise'] = p_noise
+
+        return res   
```

Comment:
Since we're adding a return, we should add documentation of it into the function doc comments